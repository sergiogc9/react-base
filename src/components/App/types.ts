import { TProps } from '@src/lib/i18n';

export type StateProps = {
	authenticated: boolean
};

export type DispatchProps = {
	onAuth: () => void
};

export type OwnProps = {};

export type ComponentProps = StateProps & DispatchProps & OwnProps & TProps;
