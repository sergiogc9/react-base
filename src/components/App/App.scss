@import "@src/scss/_globals.scss";
@import "@src/scss/_mixins.scss";
div#appLoader {
	width: 100%;
	height: 100%;
	@include flexCenter();
}

#app {
	height: 100%;
	display: flex;
	flex-direction: column;
	#appWrapper {
		height: calc(100% - #{$toolbar-height});
		display: flex;
		flex-direction: column;
	}
	#appContent {
		flex-grow: 1;
		width: 100%;
		max-width: $max-layout-width;
		background-color: white;
		align-self: center;
		overflow-y: auto;
	}
}
