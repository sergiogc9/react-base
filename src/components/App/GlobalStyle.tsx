import { createGlobalStyle } from 'styled-components';
import { reset } from '@sergiogc9/react-ui-theme';

const GlobalStyle = createGlobalStyle`
	${reset}

	html, body, #root {
		height: 100%;
	}
`;

export default GlobalStyle;
