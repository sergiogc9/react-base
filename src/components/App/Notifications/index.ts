import { connect } from 'react-redux';
import { Dispatch } from 'redux';

import { operators } from '@src/store/app/notifications';
import { State } from '@src/store/types';
import { Notification } from '@src/types/notification';
import { DispatchProps, StateProps } from './types';

import Notifications from './Notifications';

const mapStateToProps = ({ app: { notifications: { queue } } }: State): StateProps => ({
	notifications: queue
});

const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => ({
	onAddNotification: (notification: Notification) => dispatch(operators.add({ notification })),
	onUnqueueNotification: () => dispatch(operators.unqueue())
});

export default connect(mapStateToProps, mapDispatchToProps)(Notifications);
