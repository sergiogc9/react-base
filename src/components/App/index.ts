import { Dispatch } from 'redux';
import { connect } from 'react-redux';
import { operators as authOperators } from '@src/store/app/auth';
import { State } from '@src/store/types';
import { DispatchProps, StateProps } from './types';

import App from './App';

const mapStateToProps = ({ app: { auth, profile } }: State): StateProps => ({
	authenticated: auth.authenticated
});

const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => ({
	onAuth: () => dispatch(authOperators.fetchAuth())
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
