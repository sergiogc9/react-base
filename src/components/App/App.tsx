import React from 'react';
import { useSelector } from 'react-redux';
import { Helmet, HelmetProvider } from 'react-helmet-async';
import { LoadingBar } from '@sergiogc9/react-ui';

import ErrorBoundary from 'components/common/ErrorBoundary';
import FallbackLoader from 'components/common/FallbackLoader';
import uiSelectors from 'store/ui/selectors';

import NotificationsProvider from './Notifications/NotificationsProvider';
import Routes from './Routes';
import GlobalStyle from './GlobalStyle';

const App: React.FC = () => {
	const loadingBarApiCalls = useSelector(uiSelectors.getPendingLoadingBarApiCalls);

	return (
		<HelmetProvider>
			<Helmet>
				<title>SergioGCosgaya React Boilerplate</title>
			</Helmet>
			<GlobalStyle />
			<ErrorBoundary>
				<NotificationsProvider>
					<LoadingBar isVisible={loadingBarApiCalls > 0} />
					<React.Suspense fallback={<FallbackLoader />}>
						<Routes />
					</React.Suspense>
				</NotificationsProvider>
			</ErrorBoundary>
		</HelmetProvider>
	);
};

export default React.memo(App);
