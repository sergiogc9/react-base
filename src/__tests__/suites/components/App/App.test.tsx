import React from "react";
import { shallow } from "enzyme";

import App from "@src/components/App/App";

const onAuthMock = jest.fn();

describe("App", () => {

	beforeAll(() => {
		Object.defineProperty(window, "matchMedia", {
			value: jest.fn(() => ({ matches: true }))
		});
	});

	it("Not authenticated shows loader", () => {
		const wrapper = shallow(<App
			authenticated={false}
			onAuth={onAuthMock}
		/>);
		expect(onAuthMock).toHaveBeenCalled();
		expect(wrapper).toMatchSnapshot();
	});

	it("Authenticated shows routes", () => {
		const wrapper = shallow(<App
			authenticated={false}
			onAuth={onAuthMock}
		/>);
		expect(onAuthMock).toHaveBeenCalled();
		wrapper.setProps({ authenticated: true });
		expect(wrapper).toMatchSnapshot();
	});
});
