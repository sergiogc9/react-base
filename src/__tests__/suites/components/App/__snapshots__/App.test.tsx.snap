// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Authenticated shows routes 1`] = `
<div
  id="app"
>
  <Connect(Notifications) />
</div>
`;

exports[`App Not authenticated shows loader 1`] = `
<div
  id="appLoader"
>
  <img
    alt=""
    src="/appLoader.svg"
  />
</div>
`;
